homeassistant/components/sensor/nut.py

```diff
@@ -0,0 +1,291 @@
+    'battery.temperature':
+        ['Battery Temperature', TEMP_CELSIUS, 'mdi:thermometer'],
+    'battery.runtime': ['Battery Runtime', 'sec', 'mdi:timer'],
+    'battery.runtime.low': ['Low Battery Runtime', 'sec', 'mdi:timer'],
+    'battery.runtime.restart':
+        ['Minimum Battery Runtime to Start', 'sec', 'mdi:timer'],
+    'battery.alarm.threshold':
+        ['Battery Alarm Threshold', '', 'mdi:information-outline'],
+    'battery.date': ['Battery Date', '', 'mdi:calendar'],
+    'battery.mfr.date': ['Battery Manuf. Date', '', 'mdi:calendar'],
+    'battery.packs': ['Number of Batteries', '', 'mdi:information-outline'],
+    'battery.packs.bad':
+        ['Number of Bad Batteries', '', 'mdi:information-outline'],
+    'battery.type': ['Battery Chemistry', '', 'mdi:information-outline'],
+    'input.sensitivity':
+        ['Input Power Sensitivity', '', 'mdi:information-outline'],
+    'input.transfer.low': ['Low Voltage Transfer', 'V', 'mdi:flash'],
+    'input.transfer.high': ['High Voltage Transfer', 'V', 'mdi:flash'],
+    'input.transfer.reason':
+        ['Voltage Transfer Reason', '', 'mdi:information-outline'],
+    'input.voltage': ['Input Voltage', 'V', 'mdi:flash'],
+    'input.voltage.nominal': ['Nominal Input Voltage', 'V', 'mdi:flash'],
+}
+
+STATE_TYPES = {
+    'OL': 'Online',
+    'OB': 'On Battery',
+    'LB': 'Low Battery',
+    'HB': 'High Battery',
+    'RB': 'Battery Needs Replaced',
+    'CHRG': 'Battery Charging',
+    'BYPASS': 'Bypass Active',
+    'CAL': 'Runtime Calibration',
+    'OFF': 'Offline',
+    'OVER': 'Overloaded',
+    'TRIM': 'Trimming Voltage',
+    'BOOST': 'Boosting Voltage',
+    'FSD': 'Forced Shutdown',
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_ALIAS, default=None): cv.string,
+    vol.Optional(CONF_USERNAME, default=None): cv.string,
+    vol.Optional(CONF_PASSWORD, default=None): cv.string,
+    vol.Required(CONF_RESOURCES, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Setup the NUT sensors."""
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    alias = config.get(CONF_ALIAS)
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    data = PyNUTData(host, port, alias, username, password)
+
+    if data.status is None:
+        _LOGGER.error("NUT Sensor has no data, unable to setup.")
+        return
+
+    _LOGGER.debug('NUT Sensors Available: %s', data.status)
+
+    entities = []
+
+    for resource in config[CONF_RESOURCES]:
+        sensor_type = resource.lower()
+
+        if sensor_type not in SENSOR_TYPES:
```

Comment:
Thanks for the clarification balloob, I understand now and will make the changes.